lib/src/main/java/io/ably/lib/realtime/ChannelStateListener.java

```diff
@@ -37,6 +37,13 @@
 			this.reason = reason;
 			this.resumed = resumed;
 		}
+
+		/* constructor for UPDATE event */
```

Comment:
I think we could use a factory constructor here, so it can have a meaningful name instead of needing a comment like this